libraries/abstractions/common_io/test/test_scripts/uart/test_iot_uart_test.py

```diff
@@ -0,0 +1,260 @@
+        self._serial.write("\r\n".encode("utf-8"))
+
+        dut_res =  str(self._serial.read_until(terminator=serial.to_bytes([ord(c) for c in 'Ignored \n\r'])))
+        dut_result = dut_res_pattern.search(dut_res)
+        print(dut_res)
+        t_shell.join()
+
+        with open(self.rpi_output_file, "r") as f:
+            rpi_output = str(f.read());
+            print(rpi_output)
+            pi_result = pi_regex.search(rpi_output)
+
+        s.send(b'E')
+        s.close()
+        self.clean()
+
+        return "Pass" if pi_result and dut_result else "Fail"
+
+    def test_IotUartBaudChange(self):
+        """
+        Test body of uart baud change test. Return pass if expected messages are recieved
+        :return: 'Pass' or 'Fail'
+        """
+        result_matches = []
+        dut_result = None
+        dut_res_pattern = re.compile(r"AFQP_AssistedIotUARTBaudChange\W\sPASS")
+        pi_res_pattern = re.compile(r"UART\sread-write")
+
+        t_shell = threading.Thread(target=self.run_shell_script,
+                                   args=(" ".join([self.shell_script, self._ip, self._login, self._pwd, '2']),))
+        t_shell.start()
+
+        # IMPORTANT: without this, the socket won't connect on mac; reason not known.
+        sleep(1)
+
+        # Create a tcp type socket with AF_INET address family.
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        time_out = 10
+        # Wait until connection with the process on rpi is established.
+        while s.connect_ex((self._ip, self.port)) != 0 and time_out > 0:
+            time_out -= 1
+            sleep(1)
+        if time_out == 0:
+            print("Socket connection cannot be established")
+            s.close()
+            return "Fail"
+
+        print("connnected successfully")
+
+        self._serial.reset_input_buffer()
+
+        cmd = "iot_tests test 1 2"
+        self._serial.write("\r\n".encode("utf-8"))
+        self._serial.write(cmd.encode("utf-8"))
+        self._serial.write("\r\n".encode("utf-8"))
+
+        dut_res =  str(self._serial.read_until(terminator=serial.to_bytes([ord(c) for c in 'Ignored \n\r'])))
+        dut_result = dut_res_pattern.search(dut_res)
+
+        print(dut_res)
+        t_shell.join()
+
+        with open(self.rpi_output_file, "r") as f:
+            result_matches = pi_res_pattern.findall(str(f.read()))
+        s.send(b'E')
+        s.close()
+        self.clean()
+
+        return "Pass" if len(result_matches) >= 2 and dut_result else "Fail"
+
+    def test_IotUartWriteAsync(self):
+        """
+        Test body of uart write async test. Return pass if expected message is received
+        :return: 'Pass' or 'Fail'
+        """
+        pi_result = None
+        dut_result = None
+        dut_res_pattern = re.compile(r"AFQP_AssistedIotUARTWriteAsync\W\sPASS")
+        # 200 corresponds to the config testIotUART_BUFFER_LENGTH_LARGE on the DUT
+        pi_res_pattern = re.compile(r"\b(?:\\xaa){200}\b")
```

Comment:
This should be 199, as the last byte in the buffer is '\n', not 0xAA

Reply:
The size of buffer is 201: testIotUART_BUFFER_LENGTH_LARGE

Reply:
The write is only 200 bytes. https://github.com/aws/amazon-freertos/blob/2049152156e5ffc7d2d775da29b527487cc857b1/libraries/abstractions/common_io/test/test_iot_uart.c#L428

Reply:
Oh, You are right. Let me update the test to be:

```
 lWrite = iot_uart_write_async( xUartHandle, cpBufferLarge, testIotUART_BUFFER_LENGTH_LARGE + 1 );
```

Reply:
I do not see too much sense to modify c code just to keep 200 here. Let us just keep consistent with c code. It does not matter 199 or 200 here.

Reply:
I agree with @cmcgcc

Reply:
ok